### PR TITLE
add form

### PR DESCRIPTION
**Pull request by fickryiman**
- Use preprocessed HTML file with embedded Ruby code.
- Use layouts and templates for shared content.
- Create a method called current_user in your ApplicationController to make current user data available to all controllers.
   - It will return the first user from the database.
- Create forms to perform the following functions:
   - Creates a Post on behalf of the current_user (use the method that you created in your ApplicationController).
   - Create a comment on behalf of the current_user (use the method that you created in your ApplicationController).
- Allow Users to add likes to Posts.

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -1,12 +1,34 @@
...
+  def create
+    @post = current_user.posts.new(post_params)
+
+    if @post.save
+      redirect_to user_posts_path(current_user)
+    else
+      flash[:alert] = 'Something went wrong'
+      render 'new'
+    end
+  end
```
Comment (nahnahdev) [inline app/controllers/posts_controller.rb]: - [x] Great job so far @fickryiman There are a few changes I will suggest you implement and you'll be on your way to the next Milestone :100: 
- [x] When I tried creating a post for the `fourth user` as you defined the fourth user as the `current_user`, The post wasn't created however the app didn't throw any error so I'm not sure what could be causing this but kindly fix this and ensure that posts are created successfully on behalf of the current user  :wink:
- Video Reference

[Screencast from 09-30-2023 12:39:50 PM.webm](https://github.com/fickryiman/blog-app/assets/102699958/fddba217-6900-4268-b074-b06bc0f1dfb1)

File: config/routes.rb [hunk 2]
```diff
@@ -1,15 +1,19 @@
...
-  # get "/users/:id", to: "users#show", as: "user"
-  # get "/users/:user_id/posts", to:"posts#index", as: "user_posts"
-  # get "/users/:user_id/posts/:id", to: "posts#show", as: "user_post"
-  resources :users, only:[:index, :show] do
-    resources :posts, only: [:index, :show]
-  end
+  get "/users/:id", to: "users#show", as: "user"
+  get "/users", to: "users#index", as: "users"
+
+  post "/users/:user_id/posts/:id/newcomment", to: "comments#create", as: "create_new_comment"
```
Comment (nahnahdev) [inline config/routes.rb]: - [x] Great job with your implementation of the comments. However, per the project requirements, there should be a pair of routes in the `config/routes.rb` file for creating comments similar to what you did for `Posts create` and `Posts New`. Currently, you only have the `Post` route for comments in your project :+1: 
- Example
   ```ruby
      POST   /posts/:post_id/comments(.:format)     comments#create
    GET   /posts/:post_id/comments/new(.:format)     comments#new
   ```
- [x] After doing this, you should ensure that there is a separate form in the `views/comments/new.html.erb` file that accepts input for creating comments. :+1:

**Review by nahnahdev — CHANGES_REQUESTED** [review]
## Project Status: CHANGES REQUIRED :recycle: 
Hi @fickryiman , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights :fire: 
- Linters well setup and passing :heavy_check_mark: 
- Professional README :heavy_check_mark: 
- ApplicationController contains `#current_user` method :heavy_check_mark: 
- Most required routes present :heavy_check_mark: 
- Most required forms present :heavy_check_mark: 
- User is able to add a like to posts :heavy_check_mark: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

You can also consider:
- N/A

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: config/routes.rb (hunk 2 above)
Comment (fickryiman) [reply config/routes.rb]: create comment is under post controller

File: app/controllers/posts_controller.rb (hunk 1 above)
Comment (fickryiman) [reply app/controllers/posts_controller.rb]: Has successfully add new post

File: app/views/users/index.html.erb [hunk 3]
```diff
@@ -1,3 +1,16 @@
 <div class="container">
-  <%= render partial: "user", collection: @users %>
-</div>
\ No newline at end of file
+
+ <%= link_to "Add a new post", new_user_post_path(user_id: @user.id), class: "add-new-post"%>
```
Comment (rloterh) [inline app/views/users/index.html.erb:3]: - [ ] [OPTIONAL] 
Good job on making improvements! Although posts and comments can be created successfully for the first user, it appears some errors were introduced in an attempt to resolve issues from the previous review; The `Add a new post` is appearing at the show all users page and therefore it's impractical to select a specific user to add a post, unlike the screen indicated by the previous review. Let's ensure to add this button on the _users/show_  view instead.

**Add New Post button displaying on wrong page:**
image

File: app/controllers/posts_controller.rb (hunk 1 above)
Comment (rloterh) [reply app/controllers/posts_controller.rb]: Can be improved!

File: config/routes.rb (hunk 2 above)
Comment (rloterh) [reply config/routes.rb]: well resolved! ✔️

**Review by rloterh — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi Fickry,  

Your project is complete! 🏁 There is nothing else to say other than... it's time to merge it. Congratulations!  🎉💯🌟 

## Highlights 🎯 
🎲 Post and comments can be added successfully!

## Optional suggestions 🧰 
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code._

_Kindly consider my inline comments marked **[OPTIONAL]** under this review_

**You can also consider:**
Live demo link in your ReadMe is broken, kindly endeavor to fix or remove it if this app is not yet deployed.
Implementing the above change(s) will make your ReadMe stand out even better! 😉

Cheers..Happy coding! 💻 🍷 🚀 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**